### DISCUSSION TITLE
Puzzled by a panic

### DISCUSSION THREAD

**Discussion by jwwalker** [google/filament#3705] [Q&A]
When I use a material with `shadingModel : lit`, `refractionType : thin`, and `refractionMode : screenspace`, I get a fatal error:

```
utils::PostconditionPanic
in void filament::backend::metal::MetalDriver::createRenderTargetR(Handle<filament::backend::HwRenderTarget>, filament::backend::TargetBufferFlags, uint32_t, uint32_t, uint8_t, backend::MRT, filament::backend::TargetBufferInfo, filament::backend::TargetBufferInfo):248
in file filament/backend/src/metal/MetalDriver.mm
reason: The COLOR0 flag was specified, but no color texture provided.
```

If I change `refractionMode` to `cubemap` and make no other change, there is no error.  When the Material Sandbox sample is set to thin refraction with screen space refraction, there is no such error.  Any idea what I might be doing wrong?  As you can see, I'm using Metal.  It's on macOS.

**Comment by romainguy** [comment]
@bejado Any idea? I confirm that material_sandbox doesn't have the issue. It might be related to other post processing features but I can't find a combination that triggers the crash.

**Reply by romainguy** [reply]
Actually if I sync with tip of tree material_sandbox crashes on startup with Metal:

Thread 7 Crashed:: FEngine::loop
0   material_sandbox              	0x0000000108b33dd8 filament::backend::metal::MetalBuffer::getGpuBufferForDraw(id<MTLCommandBuffer>) + 40

**Reply by jwwalker** [reply]
True, I wasn't quite up to date.  After updating, material_sandbox logs this on startup:

```
-[MTLVertexDescriptorInternal newSerializedDescriptor]:726: failed assertion `Serialized Descriptor Creation
Attribute at index 2 references a buffer at index 6 that has no stride.
Attribute at index 5 references a buffer at index 7 that has no stride.
Attribute at index 6 references a buffer at index 8 that has no stride.
Attribute at index 7 references a buffer at index 9 that has no stride.
Attribute at index 8 references a buffer at index 10 that has no stride.
Attribute at index 9 references a buffer at index 11 that has no stride.
Attribute at index 10 references a buffer at index 12 that has no stride.
Attribute at index 11 references a buffer at index 13 that has no stride.
Attribute at index 12 references a buffer at index 14 that has no stride.
Attribute at index 13 references a buffer at index 15 that has no stride.
Attribute at index 14 references a buffer at index 16 that has no stride.
Attribute at index 15 references a buffer at index 17 that has no stride.
'
```

**Reply by bejado** [reply]
I'll take a look into the crash.

**Reply by bejado** [reply]
@jwwalker The crash has been fixed. Can you re-sync with main and check if your material works?

**Reply by jwwalker** [reply]
@bejado No, I still get that error if I try to use `refractionMode : screenspace`.   Not that I really need to right now, I was just experimenting.
